struct unwrap fails TypeScript's `noImplicitReturns` check

Since 1.88.0 was released, the code to unwrap a struct has failed TypeScript's `noImplicitReturns` check, which is normally a sane check to have on.

In 1.88.0, the function [`unwrapAnyValue` was added](https://github.com/stephenh/ts-proto/blob/v1.88.0/integration/struct/google/protobuf/struct.ts#L406), without an explicit return (even though you may know one of the conditionals is guaranteed to be true), resulting in:

```
src/gen/google/protobuf/struct.ts:415:4 - error TS7030: Not all code paths return a value.

415 ): string | number | boolean | Object | null | Array<any> | undefined {
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is still the case, though in a [slightly different form in the latest 1.91](https://github.com/stephenh/ts-proto/blob/v1.92.1/integration/struct/google/protobuf/struct.ts#L355), still resulting in the lint error:

```
src/gen/google/protobuf/struct.ts:397:6 - error TS7030: Not all code paths return a value.

397   ): string | number | boolean | Object | null | Array<any> | undefined {
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,0 +1,1 @@
+*.sh text eol=lf

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,2 +1,3 @@
 *.sh text eol=lf
 parameters.txt text eol=lf
+protoc-gen-dump text eol=lf

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,1 +1,2 @@
 *.sh text eol=lf
+parameters.txt text eol=lf

--- a/integration/grpc-js/google/protobuf/struct.ts
+++ b/integration/grpc-js/google/protobuf/struct.ts
@@ -353,19 +353,20 @@ export const Value = {
   },
 
   unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-    if (message.stringValue !== undefined) {
+    if (message?.stringValue !== undefined) {
       return message.stringValue;
-    } else if (message.numberValue !== undefined) {
+    } else if (message?.numberValue !== undefined) {
       return message.numberValue;
-    } else if (message.boolValue !== undefined) {
+    } else if (message?.boolValue !== undefined) {
       return message.boolValue;
-    } else if (message.structValue !== undefined) {
+    } else if (message?.structValue !== undefined) {
       return message.structValue;
-    } else if (message.listValue !== undefined) {
+    } else if (message?.listValue !== undefined) {
       return message.listValue;
-    } else if (message.nullValue !== undefined) {
+    } else if (message?.nullValue !== undefined) {
       return null;
     }
+    return undefined;
   },
 };
 

--- a/integration/simple-string-enums/google/protobuf/struct.ts
+++ b/integration/simple-string-enums/google/protobuf/struct.ts
@@ -362,19 +362,20 @@ export const Value = {
   },
 
   unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-    if (message.stringValue !== undefined) {
+    if (message?.stringValue !== undefined) {
       return message.stringValue;
-    } else if (message.numberValue !== undefined) {
+    } else if (message?.numberValue !== undefined) {
       return message.numberValue;
-    } else if (message.boolValue !== undefined) {
+    } else if (message?.boolValue !== undefined) {
       return message.boolValue;
-    } else if (message.structValue !== undefined) {
+    } else if (message?.structValue !== undefined) {
       return message.structValue;
-    } else if (message.listValue !== undefined) {
+    } else if (message?.listValue !== undefined) {
       return message.listValue;
-    } else if (message.nullValue !== undefined) {
+    } else if (message?.nullValue !== undefined) {
       return null;
     }
+    return undefined;
   },
 };
 

--- a/integration/struct/google/protobuf/struct.ts
+++ b/integration/struct/google/protobuf/struct.ts
@@ -353,19 +353,20 @@ export const Value = {
   },
 
   unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-    if (message.stringValue !== undefined) {
+    if (message?.stringValue !== undefined) {
       return message.stringValue;
-    } else if (message.numberValue !== undefined) {
+    } else if (message?.numberValue !== undefined) {
       return message.numberValue;
-    } else if (message.boolValue !== undefined) {
+    } else if (message?.boolValue !== undefined) {
       return message.boolValue;
-    } else if (message.structValue !== undefined) {
+    } else if (message?.structValue !== undefined) {
       return message.structValue;
-    } else if (message.listValue !== undefined) {
+    } else if (message?.listValue !== undefined) {
       return message.listValue;
-    } else if (message.nullValue !== undefined) {
+    } else if (message?.nullValue !== undefined) {
       return null;
     }
+    return undefined;
   },
 };
 

--- a/integration/value/google/protobuf/struct.ts
+++ b/integration/value/google/protobuf/struct.ts
@@ -353,19 +353,20 @@ export const Value = {
   },
 
   unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-    if (message.stringValue !== undefined) {
+    if (message?.stringValue !== undefined) {
       return message.stringValue;
-    } else if (message.numberValue !== undefined) {
+    } else if (message?.numberValue !== undefined) {
       return message.numberValue;
-    } else if (message.boolValue !== undefined) {
+    } else if (message?.boolValue !== undefined) {
       return message.boolValue;
-    } else if (message.structValue !== undefined) {
+    } else if (message?.structValue !== undefined) {
       return message.structValue;
-    } else if (message.listValue !== undefined) {
+    } else if (message?.listValue !== undefined) {
       return message.listValue;
-    } else if (message.nullValue !== undefined) {
+    } else if (message?.nullValue !== undefined) {
       return null;
     }
+    return undefined;
   },
 };
 

--- a/src/main.ts
+++ b/src/main.ts
@@ -1370,19 +1370,20 @@ function generateUnwrap(fullProtoTypeName: string): Code[] {
 
   if (isAnyValueTypeName(fullProtoTypeName)) {
     chunks.push(code`unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-      if (message.stringValue !== undefined) {
+      if (message?.stringValue !== undefined) {
         return message.stringValue;
-      } else if (message.numberValue !== undefined) {
+      } else if (message?.numberValue !== undefined) {
         return message.numberValue;
-      } else if (message.boolValue !== undefined) {
+      } else if (message?.boolValue !== undefined) {
         return message.boolValue;
-      } else if (message.structValue !== undefined) {
+      } else if (message?.structValue !== undefined) {
         return message.structValue;
-      } else if (message.listValue !== undefined) {
+      } else if (message?.listValue !== undefined) {
           return message.listValue;
-      } else if (message.nullValue !== undefined) {
+      } else if (message?.nullValue !== undefined) {
         return null;
       }
+      return undefined;
     }`);
   }
 

